### ISSUE TITLE
Notes in questionnaire silently carry forward and save

### ISSUE THREAD

**Issue by bsnouffer** [shcaba/FishPath-updates#289]
A note written for a question in the questionnaire section will carry forward and be saved for every subsequent question in that section with no indication given to the user.

Steps to reproduce:
1) Take a note for a question:
![image](https://user-images.githubusercontent.com/11687620/64453037-7ea92280-d09c-11e9-844a-b435de86e8a3.png)
2) Hit next question
3) Notice that the note button indicates that there is no note
![image](https://user-images.githubusercontent.com/11687620/64453075-9a142d80-d09c-11e9-8971-f8311cb15424.png)
4) Click take a note. The text from the previous question is still there:
![image](https://user-images.githubusercontent.com/11687620/64453126-b617cf00-d09c-11e9-9443-03e950b1b354.png)

Note that even if you don't open the note editor, this text still saves to the question and the user won't find out unless they hit previous, or until they view the answers page after the questionnaire is complete.

**Comment by bsnouffer** [comment]
This will also happen when hitting the previous button. 
Also bigger issue than previously known. If there is already a note for the question, it **_overwrites_** the actual note with what is carried from the prior question.

**Comment by bsnouffer** [comment]
From a quick debugging session before travel, looks like the issue is that the note state is being set in the constructor. However, the constructor is only called once when loading the entire questionnaire section. Seems like the code to initialize the text for the notes for each question needs to be moved out of the constructor and put somewhere that will be called whenever moving to a new question. 

```
class QuestionNoteInlineEditor extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      noteState: toEditorState(
        this.props.defaultValue || this.props.question.notesPrompt || ""
      )
    };
  }
```

**Comment by FishPath** [comment]
Thank you @bsnouffer for investigating this today! We can explain the bug to participants and be cautious during workshop note taking until it can be addressed.

**Comment by bsnouffer** [comment]
From Serena, experienced during recent Chile shark workshop:
“blank notes” will also carry over when you go “Previous”, deleting any previous notes!! I’m not really sure if this is actually happening but this is what it seems like. Either that or notes just not saving at all.

**Comment by bsnouffer** [comment]
May have been caused by update to React 16. See [here for explanation and solution](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-uncontrolled-component-with-a-key) (added key to notes editor component in questionpage) 

Fixed in mcclintock-lab/FishPath#59
